src/Diagnostics/atmos_core.jl

```diff
@@ -70,15 +58,41 @@ function atmos_core_collect(dgngrp::DiagnosticsGroup, currtime)
     end
     FT = eltype(localQ)
 
-    nstate = num_state(bl, FT)
-    nauxstate = num_aux(bl, FT)
-    ndiff = num_diffusive(bl, FT)
+    zvals = AtmosCollected.zvals
+
+    core_repdvsr = zeros(FT, Nqk * nvertelem)
+    thermo_array =
+        [zeros(FT, num_thermo(bl, FT)) for _ in 1:npoints, _ in 1:nrealelem]
+    ql_w_gt_0 = [zeros(FT, (Nq * Nq * nhorzelem)) for _ in 1:(Nqk * nvertelem)]
+    @visitQ nhorzelem nvertelem Nqk Nq begin
+        evk = Nqk * (ev - 1) + k
+
+        state = extract_state(dg, localQ, ijk, e)
+        aux = extract_aux(dg, localaux, ijk, e)
+        MH = localvgeo[ijk, grid.MHid, e]
 
-    # TODO
+        thermo = thermo_vars(bl, thermo_array[ijk, e])
+        compute_thermo!(bl, state, aux, thermo)
+
+        if thermo.moisture.q_liq > 0 && state.w > 0
```

Comment:
Does it make more sense to change such conditions to `!(bl.moisture isa DryModel)`?

Reply:
Just a thought

Reply:
Or is this definition valid for `bl.moisture isa DryModel`, too?

Reply:
This is totally WIP. This will end up working just like [this](https://github.com/climate-machine/CLIMA/blob/master/src/Diagnostics/thermo.jl#L44).